week3/MyPlayground.playground/Contents.swift

```diff
@@ -0,0 +1,141 @@
+
+for i in firstArray {
+  for j in secondArray {
+    var result = "\(i) * \(j) = \(i*j)"
+    print(result)
+  }
+}
+
+//d) Write a function called average that takes an optional array of Int. If the array is not nil, calculate the average of the array's values and print:
+//“The average of the values in the array is <average>.”
+//
+//Where <average> is the calculated average. If the array is nil, print:
+//“The array is nil. Calculating the average is impossible.”
+
+func average(nums: [Int]?) {
+  if let nonOptionalNums = nums {
+    if nonOptionalNums.count > 0 {
+      let total = nonOptionalNums.reduce(0, +)
+      let average = total/nonOptionalNums.count
+      print("The average of the values in the array is \(average)")
+    } else {
+      print("The array is nil. Calculating the average is impossible.")
+    }
+  }
+}
+
+average(nums: nums)
+let optionalIntArray: [Int]? = []
+average(nums: optionalIntArray)
+
+//e) Create a struct called Person with the properties firstName, lastName, and age. Choose appropriate data types for the properties. Include a method on Person called details that prints the values stored in the properties in the following format:
+// Name: <firstName> <lastName>, Age: <age>
+//Create an instance of Person called person. Pass your first name,last name, and age for the properties. Finally, call the method details.
+
+struct Person {
+  var firstName: String = ""
+  var lastName: String = ""
+  var age: Int = 0
+
+  func details() {
+    let personDetails = "Name: \(firstName) \(lastName), Age: \(age)"
+    print(personDetails)
+  }
+}
+
+var person = Person()
+person.firstName = "Jaswanth"
+person.lastName = "Alahari"
+person.age = 32
+
+person.details()
+
+// f) Create a class called Student with two properties: person of type Person and grades, an array of Int. The class must have a method called calculateAverageGrade that takes no parameters and returns a Double. Include a method called details that prints the values of the properties stored in Student along with the average grade in the following format:
+//Name: <firstName> <lastName>, Age: <age>, GPA: <average grade>.
+//
+//Create an instance of Student called student passing in your first name, last name, age, and an array of five numbers: 94, 99, 81, 100, 79. Next call the details method to output the details of the student.
+
+
+class Student {
+  var person = Person()
+  var grades = [Int]()
+
+  func calculateAverageGrade() -> Double {
+    let gpa = Double(grades.reduce(0, +)/grades.count)
+    return gpa
+  }
+
+  func details() {
+    let gpa = calculateAverageGrade()
+    let studentdetails = "\(person.details()), GPA: \(gpa)"
+    print(studentdetails)
+  }
+}
+
+var student = Student()
+student.person.firstName = "Rijutha"
+student.person.lastName = "Kona"
+student.person.age = 32
+student.grades.append(94)
+student.grades.append(99)
+student.grades.append(81)
+student.grades.append(100)
+student.grades.append(79)
+
+student.details()
+
```

Comment:
Good job using grades.append(), though you could also have just created student with a single grades array. with the values.

your calculate function though should be Double(grades.reduce(0, +)) / Double(grades.count) for increased accuracy. In your version you're returning a Double divided by an int where the GPA should be 90.6.